### PR TITLE
Add failing (ignored) linearizability test

### PR DESCRIPTION
**Pull request by justinj**

### CODE REVIEW

**Review by mjibson — APPROVED** [review]
LGTM. If you want to remove both sleeps that could probably be done by running some query in a loop until it succeeds. Should be less racey than the current stuff. But also, fine as is.

**Comment by philip-stoev** [comment]
This is cool! Thanks!

File: src/materialized/tests/sql.rs [hunk 1]
```diff
@@ -544,3 +546,99 @@ fn test_temporary_views() -> Result<(), Box<dyn Error>> {
...
+    let temp_dir = tempfile::tempdir()?;
+
+    // For each source we want to create, we spawn a thread that is constantly appending to a CSV,
+    // which we tail for the source.
+    for i in 0..NUM_FILES {
+        let path = Path::join(temp_dir.path(), format!("in{}.csv", i));
+        thread::spawn({
+            let mut file = File::create(&path)?;
+            move || {
+                for _ in 0..NUM_WRITES {
```
Comment (philip-stoev) [inline src/materialized/tests/sql.rs:584]: All the files will move in relative lockstep. Some random sleeps may cause some more interesting event interleavings.

File: src/materialized/Cargo.toml [hunk 2]
```diff
@@ -109,6 +109,7 @@ postgres-openssl = "0.5.0"
 postgres-protocol = "0.6.0"
 postgres_array = "0.11.0"
 predicates = "1.0.7"
+rand = "0.8.3"
```
Comment (benesch) [inline src/materialized/Cargo.toml:112]: I think this wants to be a dev dev?

File: src/materialized/tests/sql.rs [hunk 3]
```diff
@@ -544,3 +546,99 @@ fn test_temporary_views() -> Result<(), Box<dyn Error>> {
...
+// one named s<i> which tails the CSV, and a materialized view for each named
+// v<i>.
+//
+// N.B. this test currently fails and is ignored. TODO(justin): fix it.
+// N.B. this test also fails more reliably in release mode.
+#[test]
+#[ignore]
+fn test_linearizable() -> Result<(), Box<dyn Error>> {
+    ore::test::init_logging();
+    let config = util::Config::default();
```
Comment (benesch) [inline src/materialized/tests/sql.rs:566]: Am I crazy or does this `config` not get used?

**Review by benesch — APPROVED** [review]

File: src/materialized/tests/sql.rs (hunk 3 above)
Comment (justinj) [reply src/materialized/tests/sql.rs:566]: Oh whoops, you're right, I guess the compaction window isn't needed anyway.

File: src/materialized/Cargo.toml (hunk 2 above)
Comment (justinj) [reply src/materialized/Cargo.toml:112]: Is this not under dev deps?

**Pull request merged**